impl/graphsync.go

```diff
@@ -470,8 +470,17 @@ func (gsr *graphSyncReceiver) ReceiveMessage(
 	ctx context.Context,
 	sender peer.ID,
 	incoming gsmsg.GraphSyncMessage) {
-	gsr.graphSync().responseManager.ProcessRequests(ctx, sender, incoming.Requests())
-	gsr.graphSync().requestManager.ProcessResponses(sender, incoming.Responses(), incoming.Blocks())
+
+	requests := incoming.Requests()
+	responses := incoming.Responses()
+	blocks := incoming.Blocks()
+
+	if len(requests) > 0 {
+		gsr.graphSync().responseManager.ProcessRequests(ctx, sender, incoming.Requests())
```

Comment:
nit: any reason you're not just passing in the local variables you just read? The messages are immutable.

Reply:
because I'm a hack, it was done properly on the original version I did on my tracing branch, but wasn't copied to this stand-alone branch obviously